pandas/core/reshape/concat.py

```diff
@@ -262,7 +262,10 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
         ndims = set()
         for obj in objs:
             if not isinstance(obj, NDFrame):
-                raise TypeError("cannot concatenate a non-NDFrame object")
+                msg = ('cannot concatenate object of type "{0}";'
```

Comment:
actually lets add back Panel, they are current still valid (same below)

Reply:
Haha, ok. Should I mark it as "... pd.Panel (deprecated)" or leave out the deprecated note?

Reply:
I like that first one! (mark as deprecated)

Reply:
Should be good to go now

Reply:
Let that last message be living testament to my hubris. Forgot to escape the parens for regex and busted the build system. NOW it should be good to go.